Tests/SourceKitLSPTests/ExecuteCommandTests.swift

```diff
@@ -252,7 +252,278 @@ final class ExecuteCommandTests: XCTestCase {
 
       XCTAssertEqual(
         url.lastPathComponent,
-        "MyMacroClient_L4C2-L4C19.swift",
+        "MyMacroClient_L5C3-L5C20.swift",
+        "Failed for position range between \(positionMarker.start) and \(positionMarker.end)"
+      )
+    }
+  }
+
+  func testAttachedMacroExpansion() async throws {
+    try await SkipUnless.canBuildMacroUsingSwiftSyntaxFromSourceKitLSPBuild()
+
+    var serverOptions = SourceKitLSPServer.Options.testDefault
+    serverOptions.experimentalFeatures.insert(.showMacroExpansions)
+
+    let project = try await SwiftPMTestProject(
+      files: [
+        "MyMacros/MyMacros.swift": #"""
+        import SwiftCompilerPlugin
+        import SwiftSyntax
+        import SwiftSyntaxBuilder
+        import SwiftSyntaxMacros
+
+        public struct DictionaryStorageMacro {}
+
+        extension DictionaryStorageMacro: MemberMacro {
+          public static func expansion(
+            of node: AttributeSyntax,
+            providingMembersOf declaration: some DeclGroupSyntax,
+            in context: some MacroExpansionContext
+          ) throws -> [DeclSyntax] {
+            return ["\n  var _storage: [String: Any] = [:]"]
+          }
+        }
+
+        extension VariableDeclSyntax {
+          var isStoredProperty: Bool {
+            if bindings.count != 1 {
+              return false
+            }
+
+            let binding = bindings.first!
+            switch binding.accessorBlock?.accessors {
+            case .none:
+              return true
+
+            case .accessors(let accessors):
+              for accessor in accessors {
+                switch accessor.accessorSpecifier.tokenKind {
+                case .keyword(.willSet), .keyword(.didSet):
+                  // Observers can occur on a stored property.
+                  break
+
+                default:
+                  // Other accessors make it a computed property.
+                  return false
+                }
+              }
+
+              return true
+
+            case .getter:
+              return false
+            }
+          }
+        }
```

Comment:
I would prefer to keep the macro as simple as possible so that the long macro implementation doesn’t distract from the actual test here, which is the macro expansion. It’s totally fine if for these test cases the macro only works in the very narrow application in which we are using them in this test.

Concretely, this means that we can skip the entire `isStoredProperty` check here (because we know that all properties in our test are stored). 

Similarly, I think the `DictionaryStoragePropertyMacro` is not being used (macros should expand just fine even if the added accessor macro doesn’t exist, so we can remove it.

Reply:
I removed the `isStoredProperty` as you suggested. But, the test fails if `DictionaryStoragePropertyMacro` doesn't exist.